### PR TITLE
Lauren solution

### PR DESCRIPTION
**Pull request by lknapalolo**
Completeness: 3.5
Comfort level: 4

### CODE REVIEW

**Comment by lknapalolo** [comment]
I couldn't get the array to populate with instances of the class... hope we can go over this during our one-on-one tomorrow!

**Comment by lknapalolo** [comment]
Got it working! I was trying to set each class instance to a variable and print it within the ennumerator.... I understand now :)),

File: app.rb [hunk 1]
```diff
@@ -1,20 +1,68 @@
...
-  # When printing all apartments, under each apartment print all of its tenants
+
+ruby_apartments = apartments.map do |apartment|
+  Apartment.new(apartment[:id], apartment[:address], apartment[:monthly_rent], apartment[:square_feet])
+end
+
+ruby_tenants = tenants.map do |tenant|
+  Tenant.new(tenant[:id], tenant[:name], tenant[:age], tenant[:apartment_id])
+end
+
```
Comment (andrewsunglaekim) [inline app.rb:17]: 👍

File: app.rb [hunk 2]
```diff
@@ -1,20 +1,68 @@
...
+          tenants_apartment_id = tenant.apartment_id
+          matching_apartments = ruby_apartments.find do |apartment|
+            apartment.id == tenants_apartment_id
+        end
+
+        puts "#{tenant.name} | #{matching_apartments.address}".center(50)
+        puts "=" * 50
+        end
+
+  end
```
Comment (andrewsunglaekim) [inline app.rb:67]: 👍 !!! really awesome! One thing that could be helpful is to abstract these things into methods. In that way if you ever need to do it again, you can just call the method.

File: models/apartment.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class Apartment
+  attr_reader(:id)
```
Comment (andrewsunglaekim) [inline models/apartment.rb:2]: I really like that you made this an `attr_reader` as this sort of property should only be set at the database level. I was curious though why you decided against it for the tenant model.

File: part_one.rb [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+ruby_apartments.each do |apartment|
+  puts "=" * 50
+  puts apartment.address
+  puts "=" * 50
+  tenants_of_apartment = ruby_tenants.select{|tenant| tenant.apartment_id == apartment.id}
+  tenants_of_apartment.each do |tenant|
+    puts tenant.name
+  end
+  puts nil
+end
```
Comment (andrewsunglaekim) [inline part_one.rb:140]: :+1: great job using enumerables.

**Comment by andrewsunglaekim** [comment]
really great work on this assignment. Let me know if you have any other questions. 👍

**Pull request closed without merging**